py/sight/widgets/decision/optimizers/bayes_opt_client.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from sight.widgets.decision.base_optimizer_client import BaseOptimizerClient
+from sight.worker.worker_helper import get_action_from_textproto
+from bayes_opt import BayesianOptimization
+from bayes_opt import UtilityFunction
+
+
+# Creating Bayesian Optimizer Object locally
+def _create_bayes_opt_obj(actions):
+  bayes_obj = BayesianOptimization(
+      f=None,
+      pbounds={
+          key: (v_obj.min_value, v_obj.max_value) for key, v_obj in actions.items()
+      },
+      verbose=2,
+      allow_duplicate_points=True,
+      # random_state=1,
+  )
+  return bayes_obj
+
+
+class BayesOptOptimizerClient(BaseOptimizerClient):
+
+  def __init__(self, opt_config):
+    super().__init__()
+    self.question_label = opt_config.get('question_label_to_propose', None)
```

Comment:
An API based on an opaque dict is hard to follow. Please make these explicit arguments of __init__ and pass them in via **action_dict